tools/nd_freeze

```diff
@@ -0,0 +1,167 @@
+url_date=''
+if [ ${#chunks[0]} -eq 8 ]; then
+  url_date=${chunks[0]}
+elif [ ${#chunks[0]} -eq 4 ]; then
+  url_date="${chunks[0]}${chunks[1]}${chunks[2]}"
+elif [ ${#chunks[0]} -eq 2 ]; then
+  url_date="${chunks[2]}${chunks[0]}${chunks[1]}"
+fi
+# Pull the time from the parameter string
+url_time=''
+if [ ${#chunks[@]} -ge 2 ] && [ ${#chunks[-1]} -eq 6 ]; then
+  url_time="${chunks[-1]}"
+elif ([ ${#chunks[0]} -gt 4 ] && [ ${#chunks[@]} -eq 3 ]) || [ ${#chunks[@]} -eq 5 ]; then
+  url_time="${chunks[-2]}${chunks[-1]}00"
+elif ([ ${#chunks[0]} -gt 4 ] && [ ${#chunks[@]} -eq 4 ]) || [ ${#chunks[@]} -eq 6 ]; then
+  url_time="${chunks[-3]}${chunks[-2]}${chunks[-1]}"
+fi
+
+if [ -z "$url_date" ]; then
+cat << EOT
+ERROR: Invalid date
+
+Valid date formats include:
+  yyyymmddThh:mm:ssZ
+  yyyymmdd
+  mm/dd/yyyy
+  yyyy-mm-dd
+  yyyymmddThh:mm
+  "yyyy-mm-dd hh:mm:ss"
+  "mm/dd/yy hh:mm:ss"
+EOT
+exit 1
+fi
+
+if [ -z "$url_time" ]; then
+  previous_timestamp=$url_date
+else
+  previous_timestamp="${url_date}T${url_time}Z"
+fi
+
+# Discover distribution codename
+source_line=`cat /etc/apt/sources.list | sed -n '/^deb.*main/p' | head -1`
+codename=`echo "$source_line" | sed -E 's/^.* ([a-z]+) main.*$/\1/'`
+if [ -z "$codename" ]; then
+  echo ERROR: Unable to determine your distribution codename
+  exit 1
+fi
+
+
+# Knock on snapshot's door. This is temporarily necessary until the production
+# version of the site is made available.
+wget -O /dev/null -q http://neuro.debian.net/_files/knock-snapshots
+
+
+# Add "next" snapshot to cover package updates made after the previous snapshot
+# but before the user provided date.
+next_line=`wget -qO- http://snapshot-neuro.debian.net:5002/archive/neurodebian/${previous_timestamp}/ | sed -n '/>next<\/a>/p'`
+next_timestamp=`echo "$next_line" | sed -E 's/^.*\/archive\/neurodebian\/([0-9TZ]+)\/">next.*$/\1/'`
+if [ -z "$next_timestamp" ]; then
+  echo ERROR: Unable to determine the next snapshot timestamp.
+  exit 1
+fi
+
+# Write sources information out to /etc/apt/sources.list.d directory.
+timestamps="$previous_timestamp $next_timestamp"
+for stamp in $timestamps
+do
+cat << EOT >> $APT_SOURCE_PATH
+# NeuroDebian
+deb http://snapshot-neuro.debian.net:5002/archive/neurodebian/${stamp}/ ${codename} main contrib non-free
+deb-src http://snapshot-neuro.debian.net:5002/archive/neurodebian/${stamp}/ ${codename} main contrib non-free
+# Main Debian repository
+deb     http://snapshot.debian.org/archive/debian/${stamp}/ ${codename} main
+deb-src http://snapshot.debian.org/archive/debian/${stamp}/ ${codename} main
+# Security updates
+deb     http://snapshot.debian.org/archive/debian-security/${stamp}/ ${codename}/updates main
+deb-src http://snapshot.debian.org/archive/debian-security/${stamp}/ ${codename}/updates main
+EOT
+done
+
+echo NeuroDebian snapshot repo source information saved to $APT_SOURCE_PATH
+echo Run \"sudo apt-get update\" to make the snapshot packages available.
```

Comment:
it is ok/safe to run apt-get update right here IMHO.  it was not run within that setup script I pointed you to because that script itself was ran within/by apt process, so we could not invoke `apt-get update` due to locking.  here it should be ok IMHO